Task completed: GP-124. Ammo clip + reload

frnzee opened the pull request:
- added and implemented ammo clip class
- fixed input actions map for reloading
- added reload prefab
- added reload event

frnzee committed:
Added ammo clip and reload functionality

- added and implemented ammo clip class
- fixed input actions map for reloading
- added primitive game object for reload indicator

ega-ega commented on Assets/Scripts/Gameplay/Weapons/AmmoClip.cs:
```diff
@@ -0,0 +1,63 @@
+            }
+        }
+
+        public bool Shot()
```
Renam to waste bullet

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/AmmoClip.cs:
```diff
@@ -0,0 +1,63 @@
+
+namespace Gameplay.Weapons
+{
+    public class AmmoClip : IWeapon
```
AmmoClip dont need implemnted IWeapon

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/AmmoClip.cs:
```diff
@@ -0,0 +1,63 @@
+{
+    public class AmmoClip : IWeapon
+    {
+        private int _bulletAmount;
```
rename to _amount

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/AmmoClip.cs:
```diff
@@ -0,0 +1,63 @@
+    public class AmmoClip : IWeapon
+    {
+        private int _bulletAmount;
+        private int _maxBulletAmount;
```
rename to _size

frnzee replied:
done

frnzee committed:
Refactoring before further work

- naming fixes
- implemented ICanReload interface
- reloading indicator preparations

frnzee committed:
Refactoring

- implemented reloadbar same way as healthbar
- added reload event

ega-ega commented on Assets/Scripts/Gameplay/Weapons/AmmoClip.cs:
```diff
@@ -0,0 +1,35 @@
+            _amount = _size;
+        }
+
+        public bool HasAmmo()
```
change to property

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Character/ICanReload.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public interface ICanReload
+    {
+        float ReloadTime { get; }
```
remove from the interface

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Character/ICanReload.cs:
```diff
@@ -0,0 +1,11 @@
+    {
+        float ReloadTime { get; }
+        void Reload();
+        event Action ReloadChanged;
```
rename to Reloaded

frnzee replied:
done

frnzee committed:
Refactoring

- naming fixes
- bool HasAmmo converted to property

frnzee committed:
Refactoring

frnzee committed:
Merge branch 'develop' into feature/GP-124_Clip+reload

ega-ega commented on Assets/Scripts/Gameplay/Weapons/IWeapon.cs:
```diff
@@ -2,6 +2,8 @@ namespace Gameplay.Weapons
     public interface IWeapon : IWeaponReadonly
     {
-        bool Shot();
+        bool WasteBullet();
```
Rename to Shot() please restore

frnzee replied:
done

frnzee committed:
Conflicts resolve

ega-ega commented on Assets/Scripts/Gameplay/Character/Character.cs:
```diff
@@ -26,6 +26,8 @@ public class Character : MonoBehaviour, ICharacter, IDamageable, IHaveHealth, IC
         public float MoveSpeed => _stats.MoveSpeed;
         
         public float AttackSpeed => _stats.AttackSpeed;
+        public float ReloadTime { get; }
```
Pls remove

frnzee replied:
done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/Weapon.cs:
```diff
@@ -53,13 +56,14 @@ private void Update()
-            if (_shotCooldownTimer > 0 || _bulletAmount == 0)
+            if (_shotCooldownTimer > 0 || !_ammoClip.HasAmmo)
             {
+                Debug.Log("Need reload");
```
Pls Remove

frnzee replied:
done

frnzee committed:
Refactoring

- auto reload moved to weapon class
- removed debug info and unused variable
